New sidebar styles

thisisdano opened the pull request:
In bringing the background to the new, lighter level, I needed to redesign the sidebar to match. This  PR is paired with PRs in `cg-site` (https://github.com/18F/cg-site/pull/9 - for the homepage), `cg-docs` (https://github.com/18F/cg-docs/pull/475 -for docs), and `cg-dashboard` (https://github.com/18F/cg-dashboard/pull/685 - for the dashboard).

It builds off the additional style work in https://github.com/18F/cg-style/pull/168, so please check that out if you haven't already.

---

We've updated to new typefaces and we're at the beginning of updating the dashboard to the styles shown in our [Invision](https://gsa.invisionapp.com/share/838UY596S#/191691942_App-Overview). For reference, this is how that looks:

![screen shot 2016-10-20 at 1 06 10 pm](https://cloud.githubusercontent.com/assets/11464021/19575987/1dd9a730-96c6-11e6-9e8c-f2bc90611710.png)

---

I've updated docs and dashboard to match this basic look and feel:

**docs**

![screen shot 2016-10-20 at 5 01 13 pm](https://cloud.githubusercontent.com/assets/11464021/19582115/ec676978-96e6-11e6-8c15-0f56a7f67cd1.png)

![cg-docs-responsive 2x](https://cloud.githubusercontent.com/assets/11464021/19582295/177e99dc-96e8-11e6-9e56-b17a53204c1a.png)

**Dashboard**

![screen shot 2016-10-20 at 1 03 40 pm](https://cloud.githubusercontent.com/assets/11464021/19576154/ca1b07a0-96c6-11e6-8dc6-9e17aa18f9b1.png)

---

I've also added `<section> <h1>`s to all the docs content to give them consistent titles, and I've modified their hed structure when the body heds were out of whack.

This new functionality and look-and-feel works pretty well in all my testing, but obvs it's never perfect. This is a baseline for further work, so I'd be happy to talk about what's here, adjust it now, and look to improve it over further iterations!

It crosses over into Skyporter world as well, so all are tagged here!

@msecret @vz3 @nikzei @berndverst @brittag @jameshupp

thisisdano committed:
added missing fonts...?

Commit:
Update typography

Had to make some tweaks to make the typography acceptable in current
design.

thisisdano committed:
merged in marcos stuff

thisisdano committed:
changed formatting of stack slightly

Commit:
Fix linting errors.

thisisdano committed:
re-org of font override variables

thisisdano committed:
added mac system font to the cascade

Commit:
Fix corrupted font files

The files weren't working, probably based on where I downloaded
them from github. Made sure to download raw files over again and
things are working again.

thisisdano committed:
style changes to adapt to docs pages; dashboard fixes tk

thisisdano committed:
starting on tightening dashboard styles

thisisdano committed:
fixed up search button

thisisdano committed:
working pass on search; missing close box

thisisdano committed:
tweaks for tab nav and page-lede items

thisisdano committed:
rough implementation of page-dek element

thisisdano committed:
first linting pass

thisisdano committed:
fixed post-lint position error

thisisdano committed:
fixed close box

thisisdano committed:
fixed linting error

thisisdano committed:
changed base vars to grid

thisisdano committed:
updated font-weight: 200 to 300 for light weight

thisisdano committed:
further tweaks to vars

thisisdano committed:
fixed icons on landing

thisisdano committed:
changes for landing page

thisisdano committed:
fixed lint errors

thisisdano committed:
removed big logo variant, made default bigger

thisisdano committed:
removed big logo variant, made default bigger

thisisdano committed:
removed that debug log

thisisdano committed:
varable changes; fixed trans pngs and svgs

thisisdano committed:
tighten up form styles

thisisdano committed:
fixed logo container width

thisisdano committed:
billboard hed padding

thisisdano committed:
tweaking the disclaimer

thisisdano committed:
new styles for cg home

thisisdano committed:
new files for the new design

thisisdano committed:
merged in new origin master

thisisdano committed:
re-org vars

thisisdano committed:
fixed lint and disclaimer display

thisisdano committed:
reinstated disappearing disclaimer for responsiveness

thisisdano committed:
prelim changes for sidenav refresh

thisisdano committed:
Merge branch 'master' of github.com:18F/cg-style

thisisdano committed:
updated overwrite vars to current

thisisdano committed:
resolved conflict in billboard

thisisdano committed:
resolved conflict in actions

thisisdano committed:
resolved conflict in typography

thisisdano committed:
resolved conflict in incon-hex

thisisdano committed:
first pass at sidenav padding/margins

thisisdano committed:
fixed header logo size/margin/padding

thisisdano committed:
first pass at highlight bar

thisisdano committed:
first pass at level three display

thisisdano committed:
footer and hed spacing

thisisdano committed:
capitalization and h1 changes in docs

thisisdano committed:
scoped dek style to docs

thisisdano committed:
adjustments for basic responsive sidenav

thisisdano committed:
fixed alignment in header nav

thisisdano committed:
fixed lint errors

thisisdano committed:
tweaks to size and display of subnav

thisisdano committed:
fixed cursor and color of non-action sidebar element

thisisdano committed:
fixed non-active arrow color

thisisdano committed:
fixed lint

thisisdano committed:
changed highlight to black for consistency and accessibility

thisisdano commented:
Updated nav highlight to black for accessibility and consistency (checked against white, but not against the light bg!)

thisisdano commented:
I also think that the blue bar next to the hamburger (in mobile) might better show the doc path (i.e. something like `introduction/overview/` in the example above, to match the nav path, but that's a more involved solution, since that path should _probably_ match the url path, too...)

![cg-docs-responsive 2x](https://cloud.githubusercontent.com/assets/11464021/19578010/9b3233f2-96ce-11e6-80f7-e153943032a4.png)

thisisdano committed:
tuned mobile body type and heds

thisisdano committed:
replaced favicon

thisisdano commented:
Added a crisp new favicon for good measure!

<img width="572" alt="screen shot 2016-10-20 at 4 16 19 pm" src="https://cloud.githubusercontent.com/assets/11464021/19581282/b3e79b32-96e0-11e6-8a46-7cd4df25c52b.png">

suprenant commented:
@thisisdano nice, had noticed the favicon was falling apart!

thisisdano committed:
changed bottom line style and margin for subnav level 3

thisisdano committed:
redefined hr style as normal line

msecret commented on src/css/base/typography.scss:
```diff
@@ -181,4 +201,22 @@ pre {
\ No newline at end of file
+}
+
+pre {
```
These rules and the `.code` one above should go in `src/css/base/formatted.scss`

msecret commented on src/css/base/typography.scss:
```diff
@@ -107,21 +119,19 @@ h5 {
-  }
 }
 
+.usa-content.content h1 + p, // scope to docs
```
I think we need a better way to scope to docs. I'll think of a good way, it'll probably be a specific content class we use on docs.

msecret commented on src/css/components/actions.scss:
```diff
@@ -22,16 +20,12 @@
-  &.action-primary {
-    font-size: $sans-s2;
-  }
+  &.action-primary { font-size: $sans-s2; }
```
Can you format these to 3 lines?

msecret commented on src/css/components/footer.scss:
```diff
@@ -14,3 +14,12 @@
   }
 }
+
+.content footer {
```
@thisisdano  What is this for? Where is this being applied to?

thisisdano replied:
This is the footer that appears at the bottom of docs content panels. 

![screen shot 2016-10-28 at 12 50 35 pm](https://cloud.githubusercontent.com/assets/11464021/19820224/293876ae-9d0d-11e6-9e73-407d818781db.png)

msecret commented on src/css/components/logo.scss:
```diff
@@ -3,5 +3,14 @@
+  width: 144px;
+}
+
+.header-title a {
```
I'd prefer to use a separate class for this, not a `a` tag selector. Like `.logo-link`.

msecret commented on src/css/components/sidenav.scss:
```diff
@@ -150,76 +146,123 @@ $color-heading:     $color-primary-alt-light;
     }
   }
+
+  .sidenav-active-dashboard {
```
Rather then naming classes after the different sites, like `dashboard`, `docs`, `site`, I'd prefer we name them more specific to what they are trying to accomplish. So in this case, it looks like we're changing the colors to black. Would `sidenav-active-black` or something like that make more sense?

thisisdano replied:
this can be deprecated (and returned to `sidenav-active`) as the functionality is now the same

thisisdano committed:
added visited style to docs body links

suprenant commented:
This has been blocked pending design review for a couple days. @vz3 please review ASAP

berndverst commented:
Big 👍  from me! This looks worlds better for the documentation. So much cleaner and more professional.

Looks great from Skyporter perspective.

thisisdano committed:
changed logo size for new treatment

thisisdano commented:
I modified the logo to be more closely aligned with login.gov.
@vz3 — all cool to continue?

- - -

![screen shot 2016-10-27 at 1 26 12 pm](https://cloud.githubusercontent.com/assets/11464021/19784037/12cad48a-9c49-11e6-94f9-1ca720185376.png)

vz3 commented:
👍 here

thisisdano committed:
added attrib styles; removed cream nest

msecret commented on src/css/base/typography.scss:
```diff
@@ -107,26 +119,25 @@ h5 {
 
-.page-dek {
-  color: $color-mediumgray;
+.usa-content.content h1 + p, // scope to docs
```
Use separate class for docs specific.

thisisdano replied:
the issue with this is that it's difficult to add special classes to the docs markdown, so it's more reliable to do this via adjacency

msecret replied:
Ah thanks for pointing that out @thisisdano, I'll look at it and consider options.

msecret commented on src/css/base/typography.scss:
```diff
@@ -159,6 +174,29 @@ blockquote {
+  }
+}
+
+.quote-attrib {
```
Consider making all of these quote component.

thisisdano replied:
sure, I can do that — I was hesitant to over-specify, but I can make it so

msecret commented on src/css/components/logo.scss:
```diff
@@ -3,5 +3,14 @@
+  width: 201px;
+}
+
+.header-title a {
```
Don't use `a` here.

thisisdano replied:
Should I just add a class to that `<a>` element in the markup?

msecret commented on src/css/components/section.scss:
```diff
@@ -64,6 +75,60 @@ $chevron-size: 4.865rem;
+    text-decoration: underline;
+  }
+
+  a:active,
```
Ensure this rule isn't copied from above.

thisisdano replied:
This is an overwrite to prevent writing a separate `a, a:visited` rule for color, but I can write it that way if desired.

msecret commented on src/css/override_components.scss:
```diff
@@ -183,3 +159,16 @@ input:focus {
   }
 }
+
+.fa {
```
@thisisdano We're currently only using fontawesome on docs, are you suggesting we use it on dashboard too? I think that requires a larger discussion.

thisisdano replied:
This should be removed — I was trying something that I don't think is going anywhere!

Commit:
Move formatted rules

To `formatted.scss` to keep typography and other files cleaner.

Commit:
Remove unused font awesome rule

Commit:
Removed unused sidenav dashboard rule.

Commit:
Move quotes to own component

To keep files like typography.scss smaller and have more modular
and smaller code. Also keeps all classnames with the same pattern.

Commit:
Change classname of logo component.

To keep consistent with how we're naming classes and allow for different
styling rules for logo wrapper and logo img.
